File: entrega1/lenguajes_adicionales/go/punto_fijo.go
package main

import (
	"fmt"
	"math"
)


func f(x float64) float64 {
    return (x * math.Exp(x) - math.Pow(x,2) - 5 * x - 3) 
} 

func g(x float64) float64 {
    return ((x * math.Exp(x) - math.Pow(x,2) - 3)/5)
}

func main() {
    var xi, tol float64 = -0.5, 0.0005 
    e := tol + 1
    x := xi 
    fx := f(x)
    var iter, n, error int = 100, 0, 0
    fmt.Printf("%4s %13s %13s %13s\n", "n", "x", "f(x)", "E")

    fmt.Printf("|%4d| %13E| %13E| %13s|\n", n, x, fx, "")

    for (e > tol && fx != 0 && n < iter - 1) {
        x = g(xi)
        fx = f(x)
        e = (x - xi) 
        if(error == 1) {
            e /= x
        }
        e = math.Abs(e) 
        xi = x
        n++
    fmt.Printf("|%4d| %13E| %13E| %13E|\n", n, x, fx, e)
    }

    fmt.Printf("\nLa raiz es de f(x) = x * exp(x) - x^2 - 5x - 3, es: %f\n", x)
}

File: entrega1/lenguajes_adicionales/cpp/punto_fijo.cpp
#include <stdio.h>
#include <math.h>

#define f(x) (x * exp(x) - pow(x,2) - 5 * x - 3) // Definimos la funcion de la que se busca la raiz
#define g(x) ((x * exp(x) - pow(x,2) - 3)/5)

int main()
{
    double xi = -0.5, x = xi, fx = f(x), tol = 0.0005, e = tol + 1;
    int iter = 100, n = 0, error = 0;
    printf("|%4s| %10s| %10s| %10s|\n", "n", "x", "f(x)", "Error");

    printf("|%4d| %10f| %10f| %10f|\n", n, x, fx, "");

    while (e > tol && fx != 0 && n < iter - 1) {
        x = g(xi);
        fx = f(x);
        e = (x - xi); 
        if(error == 1) e /= x;
        e = fabs(e); 
        xi = x;
        n++;
        printf("|%4d| %10f| %10f| %10f|\n", n, x, fx, e);
    }

    printf("\nLa raiz es de f(x) = x * exp(x) - x^2 - 5x - 3; es: %f\n", x);

    return 0;
}